docs/fsharp-cheatsheet.md

```diff
@@ -368,32 +488,39 @@ Single-case discriminated unions are often used to create type-safe abstractions
     let orderId = Order "12"
 
     // Use pattern matching to deconstruct single-case DU
-    let (Order id) = orderId
+    let (Order id) = orderId  // id = "12"
 
-<div id="statically-resolved-type-parameters"></div>
+See [Discriminated Unions](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/discriminated-unions) to learn more.
 
-## Statically Resolved Type Parameters
+<div id="pattern-matching"></div>
 
-A *statically resolved type parameter* is a type parameter that is replaced with an actual type at compile time instead of at run time. They are primarily useful in conjunction with member constraints.
+## Pattern Matching
 
-    let inline add x y = x + y
-    let integerAdd = add 1 2
-    let floatAdd = add 1.0f 2.0f // without `inline` on `add` function, this would cause a type error
+Patterns are a core concept that makes the F# language and other MLs very powerful.
+They are found in `let` bindings, `match` expressions, lambda expressions, and [exceptions](#exceptions).
 
-####
+Examples of pattern matching in [Collections](#collections) and [Data Types](#data-types) can be found in their corresponding sections.
+Here are some additional patterns:
 
-    type RequestA = { Id: string; StringValue: string }
-    type RequestB = { Id: string; IntValue: int }
+    match intValue with
+    | 0 -> "Zero"                  // constant pattern
+    | 1 | 2 -> "One or Two"        // OR pattern with constants
+    | x -> $"Something else: {x}"  // variable pattern; assign value to x
 
-    let requestA: RequestA = { Id = "A"; StringValue = "Value" }
-    let requestB: RequestB = { Id = "B"; IntValue = 42 }
+    match tupleValue with
+    | (_ ,3) & (x, y) -> $"{x}, 3"  // AND pattern with a constant and variable; matches 3 and assign 3 to x
+    | _ -> "Wildcard"               // underscore matches anything
 
-    let inline getId<'t when 't : (member Id: string)> (x: 't) = x.Id
+### `when` Guard Clauses
 
-    let idA = getId requestA // "A"
-    let idB = getId requestB // "B"
+In order to match sophisticated inputs, one can use `when` to create filters, or guards, on patterns:
 
-See [Statically Resolved Type Parameters (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/generics/statically-resolved-type-parameters) and [Constraints (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/generics/constraints) for more examples.
+    match x with
+    | 0 -> 0
+    | x when x < 0 -> -1
+    | x -> 1
```

Comment:
maybe mention order sensitivity here/somewhere, i.e the fact that you couldnt move it above the when

Reply:
JetBrains is able to detect if a rule might not hit. I'll add something to the pattern matching section.

Reply:
Yeah, think compiler also flags unreachable and VS does too (but I'm mainly on Rider too)
Point though is that the cheatsheet should convey that matching is total, case ordering matters, and that you don't want to give up total matching